.github/workflows/cut-release-branch.yml

```diff
@@ -125,20 +122,33 @@ jobs:
           message="New branch ${{ inputs.new_branch_name }} created"
           echo "::notice title="New branch created": $title::$message"
 
-      # TODO: need to add scripts into the adapters to automate bumping the core 
-      # dependency on the `*.latest` branch.  The script should also bump the 
-      # nightly release test branch to the new `*.latest` branch
-      # - name: "Bump dependencies via script"
-      #   # This bumps the dependency on dbt-core in the adapters
-      #   if: ${{ !contains(github.repository, 'dbt-core') }}
-      #   run: |
-      #     echo ${{ github.repository }}
-      #     echo "running some script yet to be written now"
-      #     scripts/update_dependencies.sh
-      #     git status
-      #     git add .
-      #     git commit -m "$commit_message"
-      #     git push
+      - name: "Bump dependencies via script"
```

Comment:
How will this align with the version of `dbt-core` in `setup.py`? I think the idea is that 'dbt-<adapter>` on `main` should pull in `dbt-core` on `main` via `setup.py` and then not have a requirements of `dbt-core` at all in `dev-requirements.txt`. And then `dbt-<adapter>` versions `1.x.y` should pull in `dbt-core~=1.x` in `setup.py` and again have nothing for `dbt-core` in `dev-requirements.txt`. Is that right?

Reply:
as far as I know the `setup.py` dbt-core refs installed version to which you work with where as the `dev-requirement` pointers are soly for setting up testing enviornments and we would always want to make sure the new released branch points to the equal in core as breaking things can happen between minor versions.

Reply:
@McKnight-42 is spot on.  These changes just automate the steps we take now so that we can point to branches of core as needed while developing.  All the `*.latest` branches default to the equivalent `*.latest` `dbt-core` branch.  This script won't modify the core dependency in `setup.py`.

Reply:
Fair, I might be talking more about an enhancement out of scope of this PR than a fix to this PR. I'm trying to avoid the situation where we're testing against a different branch than we think we are. If `dev-requirements.txt` has a different version of `dbt-core` than `setup.py` does, then we wind up only testing against that version (since it's hard pinned and `setup.py` is pinned to minor), and not necessarily testing against the one that actually gets released.

Reply:
Agree this is a very important consideration/conversation but also out of scope for this ticket.